### PR TITLE
Create install.sh

### PR DESCRIPTION
**Pull request by ErranMorad**
Initial upload of the script, still working on it .

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -0,0 +1,73 @@
+#!/bin/bash
+echo Install log created at this foldes you can follow commands if scrips failed .
+exec > >(tee -a install_log.log) 2>&1
+
+echo Update repos
+sudo sed 's/# deb/deb/' -i /etc/apt/sources.list
```
Comment (goooseman) [inline install.sh]: Do we really need it? Does it open nightly and non stable repos?

File: install.sh [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+sudo apt-get update
+
+echo install basic packages
+sudo apt-get install git npm git-flow openjdk-8-jdk -y
```
Comment (goooseman) [inline install.sh]: We do not need to install npm here. It will be istalled by nvm

File: install.sh [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+echo git clone trknt
```
Comment (goooseman) [inline install.sh]: We do need at all to clone this repo.
I've sent you this repo just for the guide in the README.

File: install.sh [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+git flow init
+
+echo nvm install
+nvm install
+echo replace package and app json
+sed 's/trucknet-boilerplate-typescript-react-native/trucknet-boilerplate-typescript-react-native$(date '+%Y%m%d')/g' package.json
+sed 's/TrucknetBoilerplateTypescriptReactNative/TrucknetBoilerplateTypescriptReactNative$(date '+%Y%m%d')/g' app.json
+
+
+npx react-native eject
```
Comment (goooseman) [inline install.sh]: No need in these commands.
These commands are how to use the react-native-boilerplate. We do not need it.

File: install.sh [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+npx app-icon generate
+
+echo git install jest
```
Comment (goooseman) [inline install.sh]: No need to install all this repos and libraries. They are packaged on the project level as dependencies.

File: install.sh [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+sudo apt-get install git npm git-flow openjdk-8-jdk -y
+
+echo git clone trknt 
+git clone https://github.com/trucknet-io/trucknet-boilerplate-typescript-react-native.git
+rm -rf .git
+git init
+git remote add origin
+git flow init
+
+echo nvm install
```
Comment (goooseman) [inline install.sh]: How can you do nvm install before installing nvm? Anyway, no need

File: install.sh [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+sudo apt install code
+
+echo install zsh
+sudo apt install zsh
+chsh -s $(which zsh)
+sh -c "\$(curl -fsSL https://raw.githubusercontent.com/robbyrussell/oh-my-zsh/master/tools/install.sh)"
+
+echo install Node and NVM
+curl -o- https://raw.githubusercontent.com/creationix/nvm/v0.33.11/install.sh | bash
+sed '/plugins/nvm' ~/.zshrc
```
Comment (goooseman) [inline install.sh]: This will replace `plugins` with `nvm`. And we need to include it inside the array

File: install.sh [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+nvm install
```
Comment (goooseman) [inline install.sh]: `nvm install 10.14`

File: install.sh [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+npm update -g npm
+
+echo install android studio
+sudo apt install unzip libc6:i386 libncurses5:i386 libstdc++6:i386 lib32z1 libbz2-1.0:i386
+sudo unzip path-to-android-studio-linux.zip -d /opt
```
Comment (goooseman) [inline install.sh]: Need to download it first

File: install.sh [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+rm path-to-android-studio-linux.zip
+sudo ln -s /opt/android-studio/bin/studio.sh /usr/local/bin/android-studio
+android-studio
+sudo apt install qemu-kvm sudo adduser $(whoami) kvm
```
Comment (goooseman) [inline install.sh]: Two different lines

File: install.sh (hunk 8 above)
Comment (ErranMorad) [reply install.sh]: curl -o- https://raw.githubusercontent.com/creationix/nvm/v0.33.11/install.sh | bash
After can i do 
nvm install 10.14.0 
?

File: install.sh (hunk 9 above)
Comment (ErranMorad) [reply install.sh]: I added the url and pause the script

File: install.sh (hunk 1 above)
Comment (ErranMorad) [reply install.sh]: Removed

File: install.sh (hunk 2 above)
Comment (ErranMorad) [reply install.sh]: removed

File: install.sh (hunk 3 above)
Comment (ErranMorad) [reply install.sh]: Removd

File: install.sh (hunk 4 above)
Comment (ErranMorad) [reply install.sh]: Removed

File: install.sh (hunk 5 above)
Comment (ErranMorad) [reply install.sh]: Done

File: install.sh (hunk 6 above)
Comment (ErranMorad) [reply install.sh]: Done

File: install.sh (hunk 7 above)
Comment (ErranMorad) [reply install.sh]: I dont understand what arrey

**Comment by ErranMorad** [comment]
Updated and pushed

File: install.sh (hunk 8 above)
Comment (goooseman) [reply install.sh]: yep, but before you need to put `nvm` plugin to oh-my-zsh plugins
and the do `source ~/.zshrc` to reload the env

**Comment by goooseman** [comment]
@ErranMorad ok, it can not work this way
Let's use [Ansible](https://docs.ansible.com/ansible/latest/index.html).
Yep, it will take some time to dive into a new instrument, but it will be maintainable and they have all the recipes built-in.

For example, [ansible android studio](https://github.com/wtanaka/ansible-role-android-studio), [oh-my-zsh](https://github.com/viasite-ansible/ansible-role-zsh).

**Pull request closed without merging**